build.gradle

```diff
@@ -216,33 +223,38 @@ dependencies {
   testCompile group: 'org.springframework.boot', name: 'spring-boot-starter-test', version: versions.springBoot, {
     exclude group: 'junit', module: 'junit'
   }
-
   testCompile group: 'com.typesafe', name: 'config', version: '1.3.4'
-  testCompile group: 'org.junit.jupiter', name: 'junit-jupiter-params', version: versions.junit
-  testCompile group: 'org.junit.jupiter', name: 'junit-jupiter-api', version: versions.junit
-  testCompile group: 'org.junit.jupiter', name: 'junit-jupiter-engine', version: versions.junit
   testCompile group: 'org.mockito', name: 'mockito-junit-jupiter', version: '2.27.0'
   testCompile group: 'org.assertj', name: 'assertj-core', version: '3.12.2'
   testCompile group: 'org.springframework.cloud', name: 'spring-cloud-contract-wiremock', version: '2.0.2.RELEASE'
   testCompile group: 'org.awaitility', name: 'awaitility', version: '3.1.6'
   testCompile group: 'io.rest-assured', name: 'rest-assured', version: '3.3.0'
 
-  functionalTestCompile group: 'uk.gov.hmcts.reform', name: 'document-management-client', version: '4.0.2'
+  testImplementation group: 'org.junit.jupiter', name: 'junit-jupiter', version: versions.junit
 
   integrationTestCompile sourceSets.main.runtimeClasspath
   integrationTestCompile sourceSets.test.runtimeClasspath
 
+  integrationTestImplementation group: 'org.junit.jupiter', name: 'junit-jupiter', version: versions.junit
+
   functionalTestCompile sourceSets.main.runtimeClasspath
   functionalTestCompile sourceSets.test.runtimeClasspath
+  functionalTestCompile group: 'uk.gov.hmcts.reform', name: 'document-management-client', version: '4.0.2'
+
+  functionalTestImplementation group: 'org.junit.jupiter', name: 'junit-jupiter', version: versions.junit
 
   smokeTestCompile sourceSets.main.runtimeClasspath
   smokeTestCompile sourceSets.test.runtimeClasspath
+
+  smokeTestImplementation group: 'org.junit.jupiter', name: 'junit-jupiter', version: versions.junit
 }
 
 mainClassName = 'uk.gov.hmcts.reform.bulkscan.orchestrator.Application'
 
 bootJar {
-  archiveName = 'bulk-scan-orchestrator.jar'
+  getArchiveFileName().set(provider {
```

Comment:
do we need to set it like this ? Previous one looks simpler.

Reply:
usage of `archiveName` has been deprecated and planned to be removed in next major jump. the set is copied from what it was/is doing at the moment with `archiveName = 'some.jar'`